src/cargo/ops/registry.rs

```diff
@@ -50,16 +50,28 @@ pub struct PublishOpts<'cfg> {
     pub verify: bool,
     pub allow_dirty: bool,
     pub jobs: Option<u32>,
+    pub to_publish: ops::Packages,
     pub targets: Vec<String>,
     pub dry_run: bool,
     pub registry: Option<String>,
     pub cli_features: CliFeatures,
 }
 
-pub fn publish(ws: &Workspace<'_>, opts: &PublishOpts<'_>) -> CargoResult<()> {
-    let pkg = ws.current()?;
-    let mut publish_registry = opts.registry.clone();
+pub fn publish(root_ws: &Workspace<'_>, opts: &PublishOpts<'_>) -> CargoResult<()> {
+    let specs = opts.to_publish.to_package_id_specs(root_ws)?;
+    let mut pkgs = root_ws.members_with_features(&specs, &opts.cli_features)?;
+
+    anyhow::ensure!(pkgs.len() == 1, "can only publish one package at a time.");
```

Comment:
Am I correct in reading this in that this will support many `-p` flags and and `--workspace`, but we'll just return an error if they get matched? If so, can this error be elaborated a bit as to why multiple packages aren't supported at this time? (perhaps linking to a tracking issue?)

Reply:
It won't take --workspace, but it will take multiple -p flags. Like @ehuss said this is due to an issue with crates.io, I couldn't find any tracking issue on their repo.